### ISSUE TITLE
Gradle: Hotswap for Kotlin code does not work

### ISSUE THREAD

**Issue by nlochschmidt** [jooby-project/jooby#761]
I have a gradle based jooby project with most of the app code written in Kotlin. It works quite well however unfortunately the Hotswap feature `joobyRun` is not working.

Versions: Jooby 1.1.1, Gradle 3.4, Kotlin 1.1.2-3, Java 8u121

Steps to reproduce:
1. Check out my fork of the [kotlin-starter](https://github.com/nlochschmidt/kotlin-starter) project
2. Start the jooby app with `./gradlew joobyRun`
3. Go to http://localhost:8080. You should see `Hello Kotlin!`
4. Change response of the get handler in App.kt to `"Good day $name!"`
5. Reload the page

At this point the page should still display `Hello Kotlin!` instead of `Good day Kotlin!`

Workarounds:
1. Restart the jooby instance between each change (slow and cumbersome)
2. Start a second gradle process with `./gradlew compileJava -t` to continuously recompile the code. This seems to work but I have seen race conditions with the class loader when the reload happens before the recompilation is finished.

I am also a bit confused by the log output. Why is there an entry for `**/*.class` instead of `**/*.java` and `**/*.kt`?

```
>>> jooby:run[info|Daemon worker]:   includes: [**/*.class:**/*.conf:**/*.properties:*.js:src/*.js]
>>> jooby:run[info|Daemon worker]:   excludes: []
```

**Comment by jknack** [comment]
I'm not a gradle user and don't know how to fire compilation from a single gradle process :(

> I am also a bit confused by the log output. Why is there an entry for **/*.class instead of **/*.java and **/*.kt?

In my experience [incremental] compilation works better from IDE. Ideally when you save the file, compilation takes place. IDE regenerates .class file and Jooby reload your application.

**Comment by nlochschmidt** [comment]
Thanks for the quick response. Yes, using the IDE to compile classes is an option that works, however imo there is value in a solution that's independent from a particular editor or IDE.

I think I narrowed down the relevant places in the gradle script that would need to change. First of all the Kotlin extension `.kt` should be added here: [JoobyTask.java#L102](https://github.com/jooby-project/jooby/blob/v1.1.1/jooby-gradle-plugin/src/main/java/org/jooby/run/JoobyTask.java#L102), right? 

That change alone allows a new workaround where I can manually add the Kotlin directory as java src dir to my projects build.gradle file like so:

```
sourceSets.main.java {
    srcDirs "src/main/kotlin"
}
```

While that would work, the actual solution would be to add the Kotlin main directories to the list of `watchDirs` automatically somewhere around [JoobyProject.java#L93-L94](https://github.com/jooby-project/jooby/blob/v1.1.1/jooby-gradle-plugin/src/main/java/org/jooby/run/JoobyProject.java#L93-L94). However I guess it should somehow detect if Kotlin is even enabled and there is no such thing as a `KotlinPluginConvention` afaik. So I am not sure how to progress at the moment.

**Comment by jknack** [comment]
Just realize this was opened long time ago, fixed now in #938

**Comment by nlochschmidt** [comment]
@jknack Awesome, thanks a lot
